### PR TITLE
Bug fix LP9

### PR DESCRIPTION
**Pull request by WegrzynAB**
Change to the LP9 function from the FASTCORE algorithm to allow it to work with the models with coupling constraints (where LPproblem.A is not equal to the model.S) -> see line 28.

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

*(Note: You may replace [ ] with [X] to check the box)*

### CODE REVIEW

File: test/verifiedTests/analysis/testFASTCORE/testLP9/testLP9.m [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+currentDir = cd(fileparts(which(mfilename)));
+
+% determine the test path for references
+testPath = pwd;
+
+% load the model
+model = getDistributedModel('ecoli_core_model.mat'); %For all models in the test/models folder and subfolders
+
+% load input variables for the function
+load('options.mat');
```
Comment (laurentheirendt) [inline test/verifiedTests/analysis/testFASTCORE/testLP9/testLP9.m]: May you name this file according to the guidelines?

File: test/verifiedTests/analysis/testFASTCORE/testLP9/testLP9.m [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    fprintf(' -- Running testLP9.m using the solver interface: %s ... ', solversPkgs.LP{k});
+
+    solverLPOK = changeCobraSolver(solversPkgs.LP{k}, 'LP', 0);
+
+    if solverLPOK
+        % created test data for new version
+        V_test = LP9(options.K, options.P, model, options.LPproblem, options.epsilon);
+        Sol_test = model.c'*V_test;
+    end
+    assert(isequal(Sol_original,Sol_test), '\nResults (V) are not consistent between old and new version of LP9\n')
```
Comment (laurentheirendt) [inline test/verifiedTests/analysis/testFASTCORE/testLP9/testLP9.m]: May you name the variables according to the style guidelines? For instance: `Sol_test` should read `solTest`

File: test/verifiedTests/analysis/testFASTCORE/testLP9/testLP9.m (hunk 1 above)
Comment (WegrzynAB) [reply test/verifiedTests/analysis/testFASTCORE/testLP9/testLP9.m]: I resolved it by adding all the referenceData and input variables into one file "testData_LP9.mat"

File: test/verifiedTests/analysis/testFASTCORE/testLP9/testLP9.m (hunk 2 above)
Comment (WegrzynAB) [reply test/verifiedTests/analysis/testFASTCORE/testLP9/testLP9.m]: done

**Comment by laurentheirendt** [comment]
The test is still failing on Windows. May you have a look @WegrzynAB ? Thanks.

**Comment by WegrzynAB** [comment]
Hmm ok. Is there a specific error, or is the test itself failing (results not consistent)?

**Comment by laurentheirendt** [comment]
Let's try again.

**Comment by WegrzynAB** [comment]
I am a bit confused with this now. As the errors don't seem to be reproducible :/  

> 13:17:03 The following tests failed:
13:17:03 testAddKeyToKnownHosts.m
13:17:03 
13:17:03 The reasons were as follows:
13:17:03 ------------------------------------------------
13:17:03 testAddKeyToKnownHosts.m:
13:17:03 Error using addKeyToKnownHosts (line 54)
13:17:03  > github.com could not be added to the known hosts file in ~/.ssh/known_hosts. 
13:17:03 
13:17:03 Error in testAddKeyToKnownHosts (line 40)
13:17:03     statusAddKey = addKeyToKnownHosts();

**Comment by WegrzynAB** [comment]
I can run the test suite locally on one of the Windows computers I have access to. Perhaps I can localize the error. Let me know if you have any other ideas.

**Comment by laurentheirendt** [comment]
That error is unrelated to your code. All good - thanks 👍

**Comment by WegrzynAB** [comment]
OK!

**Comment by laurentheirendt** [comment]
Great! Well done, @WegrzynAB 👍

**Pull request merged**